tools/castle-editor/code/formproject.pas

```diff
@@ -938,6 +1099,15 @@ procedure TProjectForm.MenuItemDesignNewNonVisualClick(Sender: TObject);
     NewDesign(TCastleComponent, nil);
 end;
 
+procedure TProjectForm.MenuItemEnableDisableDockingClick(Sender: TObject);
+begin
+  IsDockUIEnabledRequested := True;
```

Comment:
The interaction between `IsDockUIEnabledRequested`, `IsDockUIEnabledChanged` is not obvious, here and in the `TProjectForm.FormDestroy`. I mean I see it's correct, but it can be simpler :) You should need just one boolean, indicating "desired" state by user, not 2 booleans (indicating whether there's a change, and if yes -> to what).

Whether it's a change -- can be always detected comparing it with current state (which you have in `IsDockUIEnabled`). Although you don't really even need this knowledge, I see.

Also you cannot toggle now between docked/non-docked without closing the window. I.e. clicking 2 times on the `MenuItemEnableDisableDocking` doesn't "toggle 2 times" (doing nothing). 

Also `IsDockUIEnabled` is a bit long (many words name) and in CGE we generally avoid using `Is` prefix for Booleans.

So:

- Rename `IsDockUIEnabled` -> `Docking`

- Remove the `IsDockUIEnabledRequested`, `IsDockUIEnabledChanged`, and instead add `WantedDocking` . 

- On loading, set `WantedDocking := Docking`.

- in the menu item `TProjectForm.MenuItemEnableDisableDockingClick`, do `WantedDocking := not WantedDocking`. This makes it obvious that the menu item toggles a boolean value.

- also synchronize the menu checkbox, doing `MenuItemEnableDisableDocking.Checked := WantedDocking` (both in `TProjectForm.MenuItemEnableDisableDockingClick` and at loading, i.e. always when you potentially change `WantedDocking`).

- in `TForm.Destroy`

```
  { Only save ProjectForm_DockUI when close the application, to make sure
    dock ui disable/enable only take effect after application restart }
  if IsDockUIEnabledRequested then
    UserConfig.SetValue('ProjectForm_DockUI', IsDockUIEnabledChanged);
```

-> can be now simplified, and I think it doesn't even need any comment, the `WantedXxx` seems natural:

```
  UserConfig.SetValue('ProjectForm_DockUI', WantedDocking);
```

- Also call it `'ProjectForm_Docking'` in settings, not `'ProjectForm_DockUI'`, to be consistent with the new name I invented.